packages/react/src/components/MenuButton/utils.js

```diff
@@ -36,50 +36,62 @@ export const getMenuPosition = ({ label, buttonRef, onPrimaryActionClick, langDi
   if (menuRect) {
     switch (overflow) {
       case 'top-right-bottom':
-        y = buttonY;
         x = isSplitButton ? buttonX : buttonX;
+        y = buttonY;
         break;
       case 'top-bottom-left':
         x = isSplitButton ? buttonX + buttonWidth + primaryButtonWidth : buttonX + buttonWidth;
         y = buttonY;
         break;
       case 'top-bottom':
-        y -= buttonHeight;
         x = isSplitButton ? buttonX + buttonWidth + primaryButtonWidth : buttonX + buttonWidth;
+        y -= buttonHeight;
         break;
       case 'top-left':
+        x = isSplitButton ? (isRtl ? buttonX : buttonX - primaryButtonWidth) : buttonX;
         y = buttonY + buttonHeight;
-        x = buttonX;
         break;
       case 'top-right':
-        x = isSplitButton ? buttonX + primaryButtonWidth + buttonWidth : buttonX + buttonWidth;
+        x = isSplitButton
+          ? isRtl
+            ? buttonX + buttonWidth + primaryButtonWidth
+            : buttonX - primaryButtonWidth - buttonWidth
+          : buttonX + buttonWidth;
         y = buttonY + buttonHeight;
         break;
       case 'right-bottom':
-        x = isSplitButton ? buttonX + primaryButtonWidth + buttonWidth : buttonX + buttonWidth;
+        x = isSplitButton
+          ? isRtl
+            ? buttonX + buttonWidth + primaryButtonWidth
+            : buttonX + buttonWidth
+          : buttonX + buttonWidth;
         y = buttonY - menuHeight;
         break;
       case 'bottom-left':
-        x = buttonX;
+        x = isSplitButton ? (isRtl ? buttonX : buttonX - primaryButtonWidth) : buttonX;
         y = buttonY - menuHeight;
```

Comment:
case `bottom` and `bottom-left` seem to be similar so merge the cases.

Reply:
thanks, @herleraja. so many possible conditions, I was just trying to get everything covered. Cleaned it up now.